src/Build.UnitTests/TelemetryTests.cs

```diff
@@ -165,5 +166,30 @@ public void WorkerNodeTelemetryCollection_CustomTargetsAndTasks()
 
             workerNodeTelemetryData.TasksExecutionData.Keys.ShouldAllBe(k => !k.IsFromNugetCache);
         }
+
+        [Fact]
+        public void Foo()
+        {
+            WorkerNodeTelemetryData wd = new WorkerNodeTelemetryData(
+                new Dictionary<TaskOrTargetTelemetryKey, TaskExecutionStats>()
+                {
+                    {
+                        new TaskOrTargetTelemetryKey("TaskA", false, true),
+                        new TaskExecutionStats(TimeSpan.FromSeconds(2), 5, 545)
+                    },
+                    {
+                        new TaskOrTargetTelemetryKey("TaskA", true, false),
+                        new TaskExecutionStats(TimeSpan.FromSeconds(254548), 6, 54545451)
+                    },
+                },
+                new Dictionary<TaskOrTargetTelemetryKey, bool>()
+                {
+                    { new TaskOrTargetTelemetryKey("TargetA", false, true, false), false },
+                    { new TaskOrTargetTelemetryKey("TargetA", true, true, false), false },
+                    { new TaskOrTargetTelemetryKey("TargetB", false, false, true), false }
+                });
+
+            var holder = TelemetryDataUtils.AsActivityDataHolder(wd);
+        }
```

Comment:
so this test is supposed to validate the serialization and you left it as a follow-up to me? did you have other tests in mind?

Reply:
That's correct - tests are completely TBD

Reply:
opened a ticket for this